src/hardware/peaktech-607x/api.c

```diff
@@ -0,0 +1,1410 @@
+{
+	struct sr_serial_dev_inst	*serial = devc->sdi->conn;
+	enum peaktech_model		model = devc->model->model;
+	int				rc;
+
+	g_assert(!devc->acq_running);
+
+	if (devc->dirty & BIT(PEAKTECH_DIRTY_CHAN_MODE)) {
+		struct pt_proto_setup_req		req =
+			pt_proto_chan_mode_req(model, devc->chan_mode);
+
+		rc = pt_serial_send_setup_op("CHAN_MODE", serial, &req);
+		if (rc < 0)
+			return rc;
+
+		devc->dirty &= ~BIT(PEAKTECH_DIRTY_CHAN_MODE);
+	}
+
+	if (devc->dirty & BIT(PEAKTECH_DIRTY_OUTPUT)) {
+		struct pt_proto_setup_req		req =
+			pt_proto_output_en_req(model, devc->output_ena);
+
+		rc = pt_serial_send_setup_op("OUTPUT_EN", serial, &req);
+		if (rc < 0)
+			return rc;
+
+		devc->dirty &= ~BIT(PEAKTECH_DIRTY_OUTPUT);
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(devc->config); ++i) {
+		struct peaktech_chan_config	*cfg = &devc->config[i];
+		char				op[64];
+
+		if (cfg->dirty & BIT(PEAKTECH_DIRTY_VOLT)) {
+			struct pt_proto_setup_req		req =
+				pt_proto_volt_set_req(model, i, cfg->set.volt);
+
+			sprintf(op, "VOLT_SET@%zd", i);
+
+			rc = pt_serial_send_setup_op(op, serial, &req);
+			if (rc < 0)
+				return rc;
+
+			cfg->dirty &= ~BIT(PEAKTECH_DIRTY_VOLT);
+		}
+
+		if (cfg->dirty & BIT(PEAKTECH_DIRTY_CURR)) {
+			struct pt_proto_setup_req		req =
+				pt_proto_curr_set_req(model, i, cfg->set.curr);
+
+			sprintf(op, "CURR_SET@%zd", i);
+
+			rc = pt_serial_send_setup_op(op, serial, &req);
+			if (rc < 0)
+				return rc;
+
+			cfg->dirty &= ~BIT(PEAKTECH_DIRTY_CURR);
+		}
+	}
+
+	return SR_OK;
+}
+
+/**
+ *  Helper function to cast an integer value to double and scale it
+ */
+static GVariant *scaled_double(unsigned int v, unsigned int scale)
+{
+	double	res = v;
+
+	res /= scale;
+
+	return g_variant_new_double(res);
+}
+
+/**
+ *  libsigrok sr_dev_driver::config_get() api function
+ */
+static int peaktech_config_get(uint32_t key, GVariant **data,
+			       struct sr_dev_inst const *sdi,
+			       struct sr_channel_group const *cg)
+{
+	struct peaktech_device		*devc = sdi ? sdi->priv : NULL;
+	int				rc;
+
+	if (!devc)
+		return SR_ERR_ARG;
+
+	/* when acquisition is not running, read the values from the device */
+	if (!devc->acq_running) {
+		union pt_proto_inquire_reply	reply;
+
+		rc = pt_serial_send_inquiry(devc->model, sdi->conn, &reply);
+		if (rc < 0) {
+			sr_err("failed to run inquire: %d", rc);
+			return rc;
+		}
+
+		pt_inquiry_parse(devc, &reply);
+	}
+
+	if (!cg) {
+		/* handle global configuration options */
+		switch (key) {
+		case SR_CONF_CHANNEL_CONFIG:
+			*data = g_variant_new_string(CHANNEL_MODES[devc->chan_mode]);
+			break;
+		case SR_CONF_ENABLED:
+			*data = g_variant_new_boolean(devc->output_ena);
+			break;
+		default:
+			return SR_ERR_NA;
+		}
+	} else {
+		/* handle channel group configuration options */
+		struct sr_channel const			*ch = cg->channels->data;
+		size_t					idx = ch->index;
+		struct peaktech_chan_config const	*cfg;
+
+		g_assert(idx < devc->model->num_chan);
```

Comment:
Right before g_assert(idx):
`if(devc->model->model == PEAKTECH_MODEL_6075 && idx == 2) { idx = 1; }`

Reply:
thanks for testing it.  Fix should be

```c
		size_t					idx = ch->index / 2;
```

Code tries to find out the channel configuration and reads the index of the first channel of a channel group.  As there are two such channels per group (see _scan() function below) it must be divided by 2.

Reply:
I guess, I can refer the cfg directly in `cg->priv` without using the index....

Reply:
> thanks for testing it. Fix should be
> 
> ```c
> 		size_t					idx = ch->index / 2;
> ```
> 
> Code tries to find out the channel configuration and reads the index of the first channel of a channel group. As there are two such channels per group (see _scan() function below) it must be divided by 2.

Okay, that makes sense. I didn't know it works like this, but had some thoughts in that direction.
Let me know if I should test something more with the P6075.

Reply:
I updated the patchset; thanks again for testing.